File: packages/android/src/main/java/com/kozmos/components/Spinner/Spinner.figma.kt
package com.kozmos.components.spinner

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-261")
class KozmosSpinnerConnect {
    @FigmaProperty(FigmaType.Enum, "Size")
    val size: KozmosSpinnerSize = Figma.mapping(
        "Small" to KozmosSpinnerSize.Sm,
        "Medium" to KozmosSpinnerSize.Md,
        "Large" to KozmosSpinnerSize.Lg,
        "XLarge" to KozmosSpinnerSize.Xl
    )

    @Composable
    fun ComponentExample() {
        // The size is the component's own now, not a modifier around it: until
        // 2026-09-22 `KozmosSpinner` took none and Dev Mode showed a spinner
        // wrapped in a `Modifier.size`, which is not how a caller should write
        // one.
        KozmosSpinner(size = size)
    }
}

File: packages/ios/Sources/Components/Spinner/Spinner.figma.swift
import SwiftUI
import Figma

struct KozmosSpinnerConnect: FigmaConnect {
    let component = KozmosSpinner.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-261"

    @FigmaEnum(
        "Size",
        mapping: [
            "Small": "sm",
            "Medium": "md",
            "Large": "lg",
            "XLarge": "xl"
        ]
    )
    var size: String = "md"

    var body: some View {
        // The size is the component's own now, not a frame around it: until
        // 2026-09-22 `KozmosSpinner` took none and Dev Mode showed a spinner
        // wrapped in a `.frame`, which is not how a caller should write one.
        KozmosSpinner(size: KozmosSpinnerSize(rawValue: self.size) ?? .md)
    }
}
